tests/kernel/mem_protect/userspace/src/main.c

```diff
@@ -1186,6 +1187,31 @@ void test_oops_stackcheck(void)
 	test_oops(K_ERR_STACK_CHK_FAIL, K_ERR_STACK_CHK_FAIL);
 }
 
+/**
+ * @brief Test system call handler for device driver subsystem API
+ *
+ * @details
+ * - System call handler for device driver subsystem API must check
+ *   that the driver implements the particular API being called.
+ *   If it is not implemented, will happen kernel oops.
+ * - In our case for the binded device is impossible to use
+ *   sensor_sample_fetch(), will happen error "unimplemented system call.
+ *
+ * @ingroup kernel_memprotect_tests
+ */
+void test_dev_driver_syscall_api_impl(void)
+{
+	int rc;
+
+	expect_fault = true;
+	expected_reason = K_ERR_KERNEL_OOPS;
+
+	struct device *dev = device_get_binding(
+			DT_CHOSEN_ZEPHYR_FLASH_CONTROLLER_LABEL);
+
+	rc = sensor_sample_fetch(dev);
```

Comment:
So, will the test case fail?

My understanding is that failure of calling sensor_sample_fetch() is expected result. The test case should be passed when that error message shows up.

Reply:
Test case will fail, if it is fail, it passed.